apps/microsoft-teams/frontend/src/components/config/ChannelSelectionModal/ChannelSelectionModal.tsx

```diff
@@ -22,23 +31,58 @@ interface ChannelSelectionModalProps {
 
 const ChannelSelectionModal = (props: ChannelSelectionModalProps) => {
   const { isShown, onClose, savedChannel, handleNotificationEdit, channels, error } = props;
+  const [searchQuery, setSearchQuery] = useState<string>('');
   const [selectedChannel, setSelectedChannel] = useState<TeamsChannel>(
     savedChannel ?? defaultNotification.channel
   );
-  const { title, link, emptyContent, emptyHeading, errorMessage } = channelSelection.modal;
+  const {
+    title,
+    link,
+    emptyContent,
+    emptyHeading,
+    errorMessage,
+    searchPlaceholder,
+    description,
+    button,
+  } = channelSelection.modal;
 
   useEffect(() => {
     const foundChannel = channels.find((channel) => channel.id === savedChannel.id);
     setSelectedChannel(foundChannel ?? defaultNotification.channel);
   }, [savedChannel]);
 
-  const SupplementalModalContent = ({ children }: { children: React.ReactNode }) => (
-    <ChannelSelectionSupplementalModalContent
-      onClose={onClose}
-      handleNotificationEdit={handleNotificationEdit}
-      selectedChannel={selectedChannel}>
-      {children}
-    </ChannelSelectionSupplementalModalContent>
+  /**
+   * The search input and the list of content types must be kept separate so that the rerendering
+   * of the list does unnecessarily re-render input (and lose focus/state). So we are maintaining
+   * search query in this component that is received from <DebouncedSearchInput> and passing it
+   * to <SearchableList>
+   */
+  const handleSearchQueryUpdate = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+  };
+
+  const renderRow = useCallback(
+    (channel: TeamsChannel) => {
+      return (
+        <Table.Row
+          key={channel.id}
+          onClick={() => setSelectedChannel(channel)}
+          className={styles.tableRow}>
+          <Table.Cell>
+            <Radio
+              id={channel.id}
+              isChecked={selectedChannel.id === channel.id}
+              onChange={() => {
+                /* clicking entire row checks this radio, i.e. do nothing here */
```

Comment:
i guess I missed this on the last PR, but curious why we need to even include the empty function. Is the `onChange` attribute reuqired or something?

Reply:
Yep, onChange is a required prop, so placating Forma-36.